test/calico/metadata_access_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build packet aws aws_edge
+// +build e2e
+
+package calico_test
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/yaml"
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+)
+
+const (
+	// Define manifest as YAML and then unmarshal it to Go stuct, so it is easier to
+	// write and debug, as it can be copy-pasted to the YAML file and applied manually.
+	metadataAccessPodManifest = `
+apiVersion: v1
+kind: Pod
+metadata:
+  generateName: metadata-access-test-
+spec:
+  restartPolicy: Never
+  containers:
+  - image: alpine
+    name: foo
+    command:
+    - wget
+    args:
+    - -O
+    - /dev/null
+    - --timeout
+    - "5"
+`
+
+	metadataAccessNamespaceManifest = `
+apiVersion: v1
+kind: Namespace
+metadata:
+  generateName: metadata-access-test-
+`
+
+	retryInterval = 1 * time.Second
+	timeout       = 5 * time.Minute
+)
+
+func TestNoMetadataAccessRandomPod(t *testing.T) { //nolint:funlen
+	t.Parallel()
+
+	client := testutil.CreateKubeClient(t).CoreV1()
+	nsclient := client.Namespaces()
+
+	ns := &corev1.Namespace{}
+	if err := yaml.Unmarshal([]byte(metadataAccessNamespaceManifest), ns); err != nil {
+		t.Fatalf("failed unmarshaling namespace manifest: %v", err)
+	}
+
+	p := &corev1.Pod{}
+	if err := yaml.Unmarshal([]byte(metadataAccessPodManifest), p); err != nil {
+		t.Fatalf("failed unmarshaling pod manifest: %v", err)
+	}
+
+	ns, err := nsclient.Create(ns)
+	if err != nil {
+		t.Fatalf("failed creating Namespace: %v", err)
+	}
+
+	metadataAddress := ""
+	platform := os.Getenv("PLATFORM")
+
+	switch platform {
+	case testutil.PlatformPacket, testutil.PlatformPacketARM:
+		metadataAddress = "https://metadata.packet.net/metadata"
+	case testutil.PlatformAWS:
+		metadataAddress = "http://169.254.169.254/latest/meta-data"
+	}
+
+	if metadataAddress == "" {
+		t.Fatalf("Platform %q not supported", platform)
+	}
+
+	// Set the right address to query, which is platform-dependent.
```

Comment:
```suggestion
	// Set the right address to query, which is platform dependent.
```

You need the hyphen only when the expression precedes a noun:

- "A platform-dependent address"
- "This address is platform dependent"